### ISSUE TITLE
Electron "portable" app detected as nsis installer, install fails

### ISSUE THREAD

**Issue by Billiam** [itchio/itch#2326]
When installing my app (https://billiam.itch.io/jotdown -> electron app, electron-builder, portable standalone exe) under Windows for the first time, it looks like the itch app detects it as an NSIS installer.

This causes it to be launched unexpectedly during the download process, before the "install" is complete. 

Once the app is closed, the itch app detects the install is complete, (presumably) removes the "installer", and then can't launch the app on subsequent runs.

`App -> gear -> manage -> Re-install` "fixes" the issue, which is even more strange, since it must then be treating the "installer" as a launchable application, but only during reinstall.

**Versions:**
itch @ 25.4.0
butler @ 15.17.0
itch-setup @ 1.18.0

**OS:** Windows 10 pro 1903

**Comment by Billiam** [comment]
Digging through electron-builder, windows portable builds use nsis and extracts to temp directories. Is there a way to prevent itch from treating it as an installer and launching/deleting it?

**Comment by fasterthanlime** [comment]
Hi, I investigated some here: https://github.com/itchio/itch/issues/2327

And my takeaway is that there is *no reliable way* to detect if modern NSIS-made .exe files are actual installers (which allow silent installation) or "portable executables" (which unpack somewhere and then run, but do not allow silent installation - aka what you have), so my conclusion is to simply drop support for NSIS installers.

What does that mean for you? Well, for "portable" apps it'll work as intended. Whenever an nsis installer is detected, it'll just get copied to the game's folder and when the game is run it'll run it.

What does that mean for others? Well, instead of attempting to run the installer silently, the installer will just be copied to the game's directory, and then when running it'll run that, so the game won't be technically launchable from the itch app, but hey, NSIS is making the "silent install" approach impossible by confusing, everything, so, 🤷‍♀️

---

**But more importantly**, my strong recommendation here is that you pick electron-builder's `"dir"` target.

Why?

  * the "portable" target is actually an NSIS self-extracting executable, which means all the contents of your app are already compressed, which completely defeats diffing & patching
  * the "portable" target needs to decompress before it runs, which means it's much slower to start up (and requires additional disk space, and will make third-party Antivirus  *completely freak out*)
  * the "zip" target generates a .zip, but as it turns out, most tools are *really bad* at generating .zip files. `butler push` accepts .zip files, but only as a convenience for folks. And when invalid .zips are pushed, [bad things happen](https://itch.io/t/562752/pushing-to-butler-indicates-no-problem-but-appears-to-silently-fail)

**TL;DR** just pick electron-builder's `dir` target and just pass that to butler. Everything will work just great. No NSIS trickery.

**Comment by Billiam** [comment]
> just pick electron-builder's dir target

Thanks. I'm doing that currently. 👍 

Have you considered making the install behavior configurable, instead of via detection?

**Comment by fasterthanlime** [comment]
> Have you considered making the install behavior configurable, instead of via detection?

I have, but I don't think it's a viable option.

There's currently 191K published games on itch.io. Even if we did provide a way to let developers specify "this is an installer that supports NSIS's silent install flags" (probably from the edit game page, because they're uploading a single file, so they can't just add an [app manifest](https://itch.io/docs/itch/integrating/manifest.html)), a large majority of developers wouldn't find, understand the necessity of, or use this option. Not to mention all the games that aren't currently being maintained - their page would need to be updated.

We could, of course, go through 1.5 million uploads ourselves (assuming infinite time/staff/money) and tag them, but new uploads would keep coming in, and some of them would lack proper tagging. 

I know this is how it works, because the only tagging we *require* right now is platform tagging:

![image](https://user-images.githubusercontent.com/7998310/65590500-2f635d00-df8b-11e9-820e-21f0aa663981.png)

..and even that is often missing. So I'm actually thinking of *removing* those checkboxes and making that detection-based as well. (Luckily, detecting that is much more reliable than detecting installer types).

As for app manifests, they are a thing, but they're mostly useful for "improvements" not "making it work at all". With manifests you can give launch targets pretty names, and icons, you can opt into API integration, etc. But you don't *need* one for your game to launch, the worst case scenario is that users have to pick between launch targets like in Winamp 3's case:

![image](https://user-images.githubusercontent.com/7998310/65590904-da741680-df8b-11e9-8734-5f452497683f.png)

Hopefully that answers your question!
